src/api/nep5.js

```diff
@@ -81,6 +81,61 @@ export const getTokenBalance = (url, scriptHash, address) => {
     })
 }
 
+/**
+ * Get the token balance of Address from Contract
+ * @param {string} url - URL of the NEO node to query.
+ * @param {Array<string>} scriptHashArray - Array of contract scriptHashes.
+ * @param {string} address - Address to query for balance of tokens.
+ * @return {Promise<{symbol: number}>}
+ */
+export const getTokenBalances = (url, scriptHashArray, address) => {
+  const addrScriptHash = reverseHex(getScriptHashFromAddress(address))
+  const scriptArray = []
+  let sb = new ScriptBuilder()
+  scriptHashArray.forEach((scriptHash, index) => {
+    sb
+      .emitAppCall(scriptHash, 'symbol')
+      .emitAppCall(scriptHash, 'decimals')
+      .emitAppCall(scriptHash, 'balanceOf', [addrScriptHash])
+    if (
+      !scriptHashArray[index + 1] ||
+      (index > 0 && index % (MAX_TOKEN_BALANCES_PER_CALL - 1) === 0)
+    ) {
+      scriptArray.push(sb.str)
+      sb = new ScriptBuilder()
+    }
+  })
+  return scriptArray.reduce((acc, script) => {
+    return Query.invokeScript(script, false)
+      .execute(url)
+      .then(res => {
+        for (let i = 0; i < res.result.stack.length; i += 3) {
+          try {
+            const symbol = hexstring2str(res.result.stack[i].value)
+            try {
+              const decimals = parseDecimals(res.result.stack[i + 1].value)
+              acc[symbol] =
+                parseHexNum(res.result.stack[i + 2].value) /
+                Math.pow(10, decimals)
+            } catch (error) {
+              log.error(
+                `single call for ${symbol} in getTokenBalances failed with : ${error}`
+              )
+              acc[symbol] = -1
+            }
+          } catch (e) {
+            log.error(`single call in getTokenBalances failed with : ${e}`)
+          }
+        }
+        return acc
+      })
+      .catch(err => {
+        log.error(`getTokenBalances failed with : ${err.message}`)
+        throw err
+      })
+  }, {})

```

Comment:
This is pretty complicated logic that should be split into more methods. I am of the opinion that if one token fails to parse, we should fail the entire method. Else, we will tend not to notice this failure when most of the tokens return.